core/src/main/java/org/testcontainers/utility/TestcontainersConfiguration.java

```diff
@@ -145,4 +153,18 @@ private static TestcontainersConfiguration loadConfiguration() {
 
         return config;
     }
+
+    /**
+     * <p>Only for testing to workaround the limitations of static, lazy singletons enforced by lombok</p>
+     *
+     * @param directoryForGlobalProps if null, defaults to {@code System.property("user.home") }
+     */
+    @VisibleForTesting
+    public void resetConfiguration(@Nullable String directoryForGlobalProps) {
```

Comment:
please use system properties for testing, not the files

Reply:
Can you help me understand what you mean here? As far as I can tell, the only way to change these properties is to use files.

Reply:
Sorry, my bad, I was thinking about another project :D

We need to think about the testing strategy here... a public method is definitely a no go

Reply:
like my comment says, this class has serious testing limitations. `@VisibleForTesting` is a pretty common pattern here.

Reply:
we should definitely improve the testing support for this class, but we cannot afford having this method as a public method